lib/ui/shared/overflow_menu.dart

```diff
@@ -84,15 +87,65 @@ class _OverflowMenuState extends ConsumerState<OverflowMenu> {
   void _enterFullscreen() =>
       ref.read(WorkspaceState.provider.notifier).toggleFullscreen(true);
 
+  Future<bool> _showOverWriteDialog() async {
+    return await showConfirmationDialog(context) ?? false;
+  }
+
+  Future<bool> _deleteFileAndProject(CatrobatImageMetaData imageData,
+      ProjectDatabase db, IFileService fileService) async {
+    final fileName = '${imageData.name}.${imageData.format.extension}';
+
+    final result = await fileService.deleteFileInApplicationDirectory(fileName);
+    if (result is Err) {
+      Fluttertoast.showToast(msg: 'Overwriting failed!');
+      return false;
+    }
+
+    Fluttertoast.showToast(msg: 'Overwriting file...');
+
+    final oldProject = await db.projectDAO.getProjectByName(imageData.name);
+    if (oldProject != null && oldProject.id != null) {
+      await db.projectDAO.deleteProject(oldProject.id!);
```

Comment:
Let's try to not user variable! anywhere.
Even though there is no error here, it has a lot of potential for errors

Let's do 
```
final projectId = oldProject.id;
if (oldProject != null && projectId != null)
      await db.projectDAO.deleteProject(projectId);
```

Reply:
Changed to ```final oldProjectId = oldProject?.id;
    if (oldProject != null && oldProjectId != null) {
      await db.projectDAO.deleteProject(oldProjectId);
      ref.invalidate(ProjectDatabase.provider);
    }```.